presto-docs/src/main/sphinx/presto_cpp/properties.rst

```diff
@@ -0,0 +1,157 @@
+===============================
+Presto C++ Properties Reference
+===============================
+
+This section describes Presto C++ configuration properties.
+
+The following is not a complete list of all configuration and
+session properties, and does not include any connector-specific
+catalog configuration properties. For information on catalog 
+configuration properties, see :doc:`Connectors </connector/>`.
+
+.. contents::
+    :local:
+    :backlinks: none
+    :depth: 1
+
+Coordinator Properties
+----------------------
+
+Set the following configuration properties for the Presto coordinator exactly 
+as they are shown in this code block to enable the Presto coordinator's use of 
+Presto C++ workers. 
+
+.. code-block:: none
```

Comment:
In my opinion, I think we should remove all references to session properties below.  

* Firstly, some of them are incorrect (e.g. whereas the _config property_ `experimental.table-writer-merge-operator-enabled` does exist and is valid, the _session property_ is `table_writer_merge_operator_enabled`, not `experimental.table-writer-merge-operator-enabled`).
* Secondly, session properties are set per query/session, and these are all invariants, so it's best to never toggle this.  Hence, they should be set as config properties only.

Reply:
I proposed to remove references to session properties as well here https://github.com/prestodb/presto/pull/22885#discussion_r1626334112.
I agree that since these are invariants, they should be set only as config properties.
There was an argument made against this that existing doc norm is to document session properties if available.

Reply:
For the purposes of this document, it's just irrelevant that a corresponding session property exists in my opinion.  In the configuration properties of our documentation we may mention that it can be toggled.  What do you think @steveburnett ?  I think just telling people that they can override this per-query is too much information, since it is **never** desirable to toggle it per-query.

Reply:
I originally made the argument to document the session property for these reasons: 

* @agrawalreetika's point "If available, we usually document the session property along with the config properties." So my first argument for including session properties was consistency with the existing doc. 
* My assumption, which may be wrong, that one or more of these configuration properties could in future be useful for a user to be able to toggle with a session property - this is a big assumption and I'll come back to it in a moment. 
If these session properties were ever to be useful to the user in future, documenting them now would improve the stability of the documentation. 

If we currently expect that the session properties for these configuration properties are never to be useful to the user, I agree they should be removed from this draft, and I can easily do that today. 
(If that current expectation changes in the future, we can always document them then.)

Reply:
For the purposes of this document, it won't ever be useful to toggle the session properties, so let's remove them.  I think it's OK to break from convention here.

Reply:
Makes sense, I'll update the PR.